### PR TITLE
Gmd/pat 313 source of intel

### PR DESCRIPTION
**Pull request by garethmdavies**

### CODE REVIEW

**Comment by garethmdavies** [comment]
The further details section has been re-coded and extracted to it's own macro. The complex code around adding the additional fields post save has been simplified and moved into the sanitise functions for now. It may be cleaner to have a dedicated service method to add the additional fields pre validation but that can be re-factored at a later date. Jquery is needed to determine when the further details is required per radio or checkbox option and also when to hide.

File: server/utils/nunjucksSetup.js [hunk 1]
```diff
@@ -51,4 +51,58 @@ module.exports = (app, path) => {
...
+
+  njkEnv.addFilter('hasFieldFurtherDetails', field => {
+    let response = false
+    field.options.forEach(option => {
+      if (option.furtherDetails) {
+        response = true
+      }
+    })
+    return response
+  })
```
Comment (yasinmustafa) [inline server/utils/nunjucksSetup.js]: the function that takes in a field and returns a response could be replaced with a more concise version:

```
field =>
    field.options.some(field => field.furtherDetails)
```

File: server/utils/nunjucksSetup.js [hunk 2]
```diff
@@ -51,4 +51,58 @@ module.exports = (app, path) => {
...
+    let response = false
+
+    answers.forEach(answer => {
+      if (answer.question === question) {
+        response = true
+      }
+    })
+    return response
+  })
 }
```
Comment (yasinmustafa) [inline server/utils/nunjucksSetup.js]: Can replace all the the field, data functions with something like :
```
(field, data) =>
  data ? field.options.filter(option => option.furtherDetails && data.answer === option.value).length : 0
```

and the respnse functions as per my previous comment

File: server/views/partials/riskBandingFurtherDetails.html [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+{% from "govuk/components/character-count/macro.njk" import govukCharacterCount %}
+
+{% macro riskBandingFurtherDetailsSection(fieldId, data, fieldType, furtherDetailsCounter ) %}
+
+    {% if fieldType == "radio" %}
+        {% set furtherDetailsClass = 'govuk-radios__conditional--hidden' if furtherDetailsCounter == 0 %}
+    {% elseif fieldType == "checkbox" %}
+        {% set furtherDetailsClass = 'govuk-checkboxes__conditional--hidden' if furtherDetailsCounter == 0%}
+    {% endif %}
+
```
Comment (timharrison-moj) [inline server/views/partials/riskBandingFurtherDetails.html]: Do we have two types of further details? I thought they were always the multi-select check-boxes?

Comment (timharrison-moj) [reply server/views/partials/riskBandingFurtherDetails.html]: Ah - its the risk-band override isn't it!? That is a radio..

**Pull request merged**